openwisp_radius/api/serializers.py

```diff
@@ -128,8 +134,47 @@ def validate(self, data):
             raise serializers.ValidationError(
                 {'number_of_users': 'The field number_of_users cannot be empty'}
             )
+        try:
+            self.batch_org = Organization.objects.get(slug=data['organization_slug'])
```

Comment:
I think this maybe a little unconventional but I save `self.batch_org` to avoid making another request to the database in the `.create()` function.
Is it fine or should I remove this variable and make another request in the `.create()`?

Reply:
question: aren't you already retrieving the organization from validating permissions?
In any case, this should be done in the view, we can access the view instance from the serializer.
See https://github.com/openwisp/openwisp-radius/blob/master/openwisp_radius/api/serializers.py#L159, the org is already retrieve in the view so the serializer just gets it from the view.
Look also for the concept of "serializer context" in DRF.

Reply:
PS: you could also add the organization to the data, eg:

```
data['organization'] = self.context['view'].organization
```
which would make the method below leaner.

Reply:
> question: aren't you already retrieving the organization from validating permissions?

Yes, but as discussed in the [BatchView permissions PR](https://github.com/openwisp/openwisp-radius/pull/131#discussion_r457825262) we don't want slug in the url and hence, I am planning to get rid of it with the [openwisp-users change PR (coming soon)](https://github.com/openwisp/openwisp-users/issues/150)

> In any case, this should be done in the view, we can access the view instance from the serializer.

Sweet, I thought it needs to be done in serializer, moving it in view would be better, I am looking into it now!